### DISCUSSION TITLE
How do I hint a filetype to rome?

### DISCUSSION THREAD

**Discussion by UtkarshVerma** [rome/tools#4420] [Q&A]
When working with files without a specific extension, e.g. `.prettierrc`. `rome` does not format it because it's an unknown extension. Is there any way I can force `rome` to format the passed file as a specific filetype, like JSON?

**Comment by ematipico** [comment]
Not at the moment, but you can work around it:

```shell
cat .prettierrc | rome format --stdin-file-path=file.json > .prettierrc
```

**Comment by juliankasimir** [comment]
@UtkarshVerma Maybe the question is aimed at a somewhat general case, but in the case of Prettier (or Eslint) you can also give the file the appropriate [file extension](https://prettier.io/docs/en/configuration.html):
like `.prettierrc.json`, `.prettierrc.js`, or `prettier.config.js`.

**Reply by UtkarshVerma** [reply]
For a personal project that works, but for contributing to open-source projects `.prettierrc` is mostly used, and I can't change the filetype.

**Reply by juliankasimir** [reply]
@UtkarshVerma Adding the extension was recommended in [Stylelint](https://stylelint.io), so we did the same for the other formatting and linting files in our project. But if that doesn't work in your case, then the idea might be as @ematipico  described above. You could possibly include this in the package.json as a normal script or if you like with [conc](https://www.npmjs.com/package/concurrently), or [lintstaged file](https://github.com/okonet/lint-staged).

[Citation](https://stylelint.io/user-guide/configure/):

> The .stylelintrc file (without extension) can be in JSON or YAML format. We recommend adding an extension to help your editor provide syntax checking and highlighting.

**Reply by ematipico** [reply]
> For a personal project that works, but for contributing to open-source projects `.prettierrc` is mostly used, and I can't change the filetype.

I'd recommend suggesting to the people that own projects with extensionless files to move out from this practice. Files without extensions are not great for DX and create a lot of friction in the sphere of developer tools. 

I don't know why and how this practice took place, but as a developer that makes developer tools, I assure you it's not easy to maintain! :)
